### ISSUE TITLE
our writers create_index() at the wrong time?

### ISSUE THREAD

**Issue by abotas** [next-exp/IC#331]
Francesc Alted asked us why we `create_index()` in our writers, as soon as we make the table, before we put anything in them. He suggests do this after we fill our pytables to avoid buggy behavior with flushes. He also says this is usually faster.

I've checked with a toy and there things seem to write 10-15% faster without the create_index. You can see that notebook [here](https://github.com/abotas/ICARO/blob/city-monitors/icaro/reco/when-to-create-index.ipynb).

I've tried to reproduce the result with irene, but so far have failed. Irene seems to be the same speed with and without `create_index`. 

Getting rid of `create_index` has the enormous advantage of avoiding unexpected crashes from pytables. Why do we have it? Should we not create it after we have filled the pytables?

**Comment by abotas** [comment]
Update: after running irene with ipmt over 1k Kr event from run 4446 a few different times, I find that Irene is consistently marginally faster without the `create_index()`. 
2.0 events/sec without `create_index`
1.9 events/sec with `create_index`

So, I think we should take it out of the writers. It's not clear to me what purpose it serves --maybe it speeds things up for cities that read the pmaps after they are written?-- so I'm sure whether we want to `create_index` after filling all the tables? or whether we want to get rid of `create_index`?

**Comment by abotas** [comment]
Update 2: 
It's putting `create_index()` _before_ filling the pytables that seems to be giving us issues. In the toy, when I `create_index()` after filling the pytables, it is faster and there is no buggy pytables behavior.

So there is no downside to putting `create_index` after filling the pytables, but I'm not sure if it does anything or if we should just get rid of it.

**Comment by gonzaponte** [comment]
`create_index` is needed for the read back of data to be fast. We can't remove, so it seems the optimal solution is to call this method *just before* closing the file.

**Comment by jjgomezcadenas** [comment]
Correct, create_index is needed and we should move it right after closing the files. Let's do it @abotas and be done with it!

**Comment by abotas** [comment]
Yesterday I wrote that simply using `create_index` after populating the pytables solved all the problems. This was not entirely true. It vastly decreases the opportunities for buggy pytables crashes, but a manual flush afterwards is still necessary. See:
https://github.com/abotas/ICARO/blob/city-monitors/icaro/reco/when-to-create-index.ipynb

Francesc Alted suggested we close the files after populating the tables, then reopen them, then index them. This completely alleviates the need to flush and prevents the pytables buggy node manager from crashing automatic flushes.

Ok, I've opened the PR so I think we should move the discussion [there](https://github.com/nextic/IC/pull/334)
